### PR TITLE
docs: add v4.0 validator upgrade guide

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #1238

This pull request started out as only adding the v4.0 validator upgrade guide and evolved to include updates to validator getting started and all upgrade guides (including the addition of the v4.0 upgrade guide).

- Added [Upgrade Guide v4.0](https://deploy-preview-1299--regen-ledger-docs.netlify.app/validators/upgrades/v4.0-upgrade.html)
- Update, simplify, and fix [Validator Get Started](https://deploy-preview-1299--regen-ledger-docs.netlify.app/validators/get-started/)
- Update, simplify, and fix [Validator Upgrades](https://deploy-preview-1299--regen-ledger-docs.netlify.app/validators/upgrades/)

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: docs/validators/upgrades/v4.0-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+## Upgrade Info
+
+### Regen Ledger v4.0
+
+- [Release Notes](https://github.com/regen-network/regen-ledger/releases/tag/v4.0.0)
+
+### Regen Mainnet
+
+- Upgrade Height: `TBD`
```
Comment (ryanchristo) [inline docs/validators/upgrades/v4.0-upgrade.md:13]: TODO before merging

File: docs/validators/upgrades/v4.0-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+### Redwood Testnet
+
+- Upgrade Height: `TBD`
```
Comment (ryanchristo) [inline docs/validators/upgrades/v4.0-upgrade.md:17]: TODO before merging

File: docs/validators/get-started/install-regen.md [hunk 3]
```diff
@@ -1,3 +1,87 @@
 # Install Regen
 
-Please see [Install Regen](../../ledger/get-started) for installation instructions.
+The following instructions are instructions for building the `regen` binary from source, also available at [Install Regen](../../ledger/get-started) alongside general information about the `regen` binary.
```
Comment (technicallyty) [inline docs/validators/get-started/install-regen.md:3]:
```suggestion
The following instructions are for building the `regen` binary from source, also available at [Install Regen](../../ledger/get-started) alongside general information about the `regen` binary.
```

**Review by technicallyty — APPROVED** [review]
utACK lgtm

**Review by aleem1314 — APPROVED** [review]
lgtm

**Pull request merged**